AmazonChimeSDK/AmazonChimeSDKTests/device/DeviceControllerTests.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  DeviceControllerTests.swift
+//  AmazonChimeSDK
+//
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+
+@testable import AmazonChimeSDK
+import AVFoundation
+import XCTest
+
+class DeviceControllerTests: XCTestCase {
+    func testListDevicesReturnsSpeakerWithCorrectType() {
+        let audioSession = MockAudioSession()
+        let videoClient = MockVideoClientController()
+        let logger = ConsoleLogger(name: "logger")
+        
+        let controller = DefaultDeviceController(audioSession: audioSession, videoClientController: videoClient, logger: logger)
+        let devices = controller.listAudioDevices()
+        let secondDevice = devices[1]
```

Comment:
Relying that it always exists on index 1 without mocking function seems little unreliable. Can we either mock the function or loop through list to find built-in speaker?

Reply:
This was referencing a hard-coded truth from `DefaultDeviceController`:
https://github.com/aws/amazon-chime-sdk-ios/blob/410c48858416efc47b7643ed62c007a5a27f075e/AmazonChimeSDK/AmazonChimeSDK/device/DefaultDeviceController.swift#L40

I suppose we can split those into 2 separate tests, one testing the position of the built-in device, and one testing the correct type. Then we can safely change this test to loop&find. Would you prefer I split those two tests?

Reply:
My bad. I think it is okay to keep it this way.